src/3d/qgspointcloudlayerchunkloader_p.cpp

```diff
@@ -42,52 +45,75 @@
 
 ///@cond PRIVATE
 
-QgsPointCloud3DGeometry::QgsPointCloud3DGeometry( Qt3DCore::QNode *parent, const QgsPointCloud3DSymbolHandler::PointData &data )
+QgsPointCloud3DGeometry::QgsPointCloud3DGeometry( Qt3DCore::QNode *parent, const QgsPointCloud3DSymbolHandler::PointData &data, QgsPointCloud3DSymbol *symbol )
   : Qt3DRender::QGeometry( parent )
   , mPositionAttribute( new Qt3DRender::QAttribute( this ) )
-  , mClassAttribute( new Qt3DRender::QAttribute( this ) )
+  , mParameterAttribute( new Qt3DRender::QAttribute( this ) )
+  , mColorAttribute( new Qt3DRender::QAttribute( this ) )
 #if QT_VERSION < QT_VERSION_CHECK(5, 10, 0)
   , mVertexBuffer( new Qt3DRender::QBuffer( Qt3DRender::QBuffer::VertexBuffer, this ) )
 #else
   , mVertexBuffer( new Qt3DRender::QBuffer( this ) )
 #endif
+  , mRenderingStyle( symbol->renderingStyle() )
 {
+  unsigned int byte_stride = mRenderingStyle == QgsPointCloud3DSymbol::RenderingStyle::RGBRendering ? 28 : 16;
   mPositionAttribute->setAttributeType( Qt3DRender::QAttribute::VertexAttribute );
   mPositionAttribute->setBuffer( mVertexBuffer );
   mPositionAttribute->setVertexBaseType( Qt3DRender::QAttribute::Float );
   mPositionAttribute->setVertexSize( 3 );
   mPositionAttribute->setName( Qt3DRender::QAttribute::defaultPositionAttributeName() );
   mPositionAttribute->setByteOffset( 0 );
-  mPositionAttribute->setByteStride( 16 );
+  mPositionAttribute->setByteStride( byte_stride );
 
-  mClassAttribute->setAttributeType( Qt3DRender::QAttribute::VertexAttribute );
-  mClassAttribute->setBuffer( mVertexBuffer );
-  mClassAttribute->setVertexBaseType( Qt3DRender::QAttribute::Float );
-  mClassAttribute->setVertexSize( 1 );
-  mClassAttribute->setName( "cls" );
-  mClassAttribute->setByteOffset( 12 );
-  mClassAttribute->setByteStride( 16 );
+  mParameterAttribute->setAttributeType( Qt3DRender::QAttribute::VertexAttribute );
+  mParameterAttribute->setBuffer( mVertexBuffer );
+  mParameterAttribute->setVertexBaseType( Qt3DRender::QAttribute::Float );
+  mParameterAttribute->setVertexSize( 1 );
+  mParameterAttribute->setName( "vertexParameter" );
+  mParameterAttribute->setByteOffset( 12 );
+  mParameterAttribute->setByteStride( byte_stride );
 
   addAttribute( mPositionAttribute );
-  addAttribute( mClassAttribute );
+  addAttribute( mParameterAttribute );
 
+  if ( mRenderingStyle == QgsPointCloud3DSymbol::RenderingStyle::RGBRendering )
```

Comment:
Again, I'd suggest moving this to a virtual method in QgsPointCloud3DSymbol, eg.

    QgsPointCloud3DSymbol::addRenderAttributes( Qt3DCore::QNode* destination );